src/AzureAuth/Ado/AdoToken.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Ado
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Microsoft.Office.Lasso.Interfaces;
+
+    /// <summary>
+    /// Business logic for either getting an ADO PAT from <see cref="IEnv"/>
+    /// or doing normal AAD Authentication.
+    /// </summary>
+    public static class AdoToken
+    {
+        private static readonly IEnumerable<string> PatEnvVars = new[]
+        {
+            EnvVars.AdoPat,
+            EnvVars.SystemAccessToken,
+        };
+
+        /// <summary>
+        /// Get either an ADO PAT from the environment. If one is not set, then do normal ADO AAD Auth.

```

Comment:
This PR has not implemented doing the normal ADO AAD Auth, right? This is just checking if env vars are set or not, then returns if any of them are set?

Reply:
I noticed the same thing. Maybe future code will change this, but for now I think this comment is incorrect. It just gets a PAT or `null`.

Reply:
This is specified in the PR description.

Reply:
Yep, was a bit confusing from the summary comment.